keanu-project/src/test/java/io/improbable/keanu/util/dot/WriteDotTest.java

```diff
@@ -0,0 +1,166 @@
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.ConstantIntegerVertex;
+import org.apache.commons.io.FileUtils;
+import org.junit.*;
+import org.nd4j.util.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+public class WriteDotTest {
+
+    private static BayesianNetwork gaussianNet;
+    private static BayesianNetwork booleanNet;
+    private static String testFileDir = "dottestdir";
+    private static String gaussianDotOutput;
+    private static String boolDotOutput;
+    private static String vertexDegree1DotOutput;
+    private static String vertexDegree2DotOutput;
+    private static String boolVertexLabel = "/BoolResult";
+    private static GaussianVertex gaussianV;
+    private static DoubleVertex doubleV;
+
+    @BeforeClass
+    public static void setup() throws IOException {
+
+        String gaussianDotFileName = testFileDir + "/gaussianDot.dot";
+        String boolDotFileName = testFileDir + "/boolDot.dot";
+        String vertexDegree1DotFileName = testFileDir + "/VertexOutput1.dot";
+        String vertexDegree2DotFileName = testFileDir + "/VertexOutput2.dot";
+
+        double mu = 0;
+        double sigma = 1;
+        gaussianV = new GaussianVertex(mu, sigma);
+        doubleV = new ConstantDoubleVertex(5.43);
+        DoubleVertex result = gaussianV.multiply(doubleV);
+        gaussianNet = new BayesianNetwork(result.getConnectedGraph());
+        WriteDot.outputDot(gaussianDotFileName, gaussianNet);
+        gaussianDotOutput = new String(Files.readAllBytes(Paths.get(gaussianDotFileName)), Charset.defaultCharset());
+
+
+        ConstantBoolVertex boolV = new ConstantBoolVertex(true);
+        ConstantIntegerVertex intV = new ConstantIntegerVertex(5);
+        DoubleVertex doubleV2 = new ConstantDoubleVertex(5.43);
+        BoolVertex comparisonV = intV.greaterThan(doubleV2);
+        BoolVertex result2 = boolV.or(comparisonV);
+        result2.setLabel(boolVertexLabel);
+        booleanNet = new BayesianNetwork(result2.getConnectedGraph());
+        WriteDot.outputDot(boolDotFileName, booleanNet);
+        boolDotOutput = new String(Files.readAllBytes(Paths.get(boolDotFileName)), Charset.defaultCharset());
+
+        WriteDot.outputDot(vertexDegree1DotFileName, gaussianV, 1);
+        vertexDegree1DotOutput = new String(Files.readAllBytes(Paths.get(vertexDegree1DotFileName)), Charset.defaultCharset());
+        WriteDot.outputDot(vertexDegree2DotFileName, gaussianV, 2);
+        vertexDegree2DotOutput = new String(Files.readAllBytes(Paths.get(vertexDegree2DotFileName)), Charset.defaultCharset());
+    }
+
+    @Test
+    public void filesGetWrittenAndAreNotOverwritten() throws IOException {
+        String fileName = "dottestfolder/test_file.dot";
```

Comment:
If you want to test all the way down to the file system then you should use Java's create temp file (`File.createTempFile(...)`). You can find an example of it in `WriteCsvTest`

Reply:
Even better would be to add the ability to write dot to an output stream that you could use to write to a test string instead of a file.

Reply:
Done